ecommerce/taxes/test/vat_rate_catalog_test.rb

```diff
@@ -0,0 +1,38 @@
+require_relative "test_helper"
+
+module Taxes
+  class VatRateCatalogTest < Test
+    class VatRateAvailableTest < VatRateCatalogTest
+      def test_returns_true_when_vat_rate_is_available
+        vat_rate = Infra::Types::VatRate.new(code: "50", rate: 50)
+        add_available_vat_rate(vat_rate)
+
+        assert catalog.vat_rate_available?(vat_rate.code)
+      end
+
+      def test_returns_false_when_vat_rate_is_not_available
+        vat_rate = Infra::Types::VatRate.new(code: "50", rate: 50)
+
+        refute catalog.vat_rate_available?(vat_rate.code)
+      end
+
+      def test_returns_false_when_vat_rate_is_not_available_even_when_other_vat_rates_are_available
```

Comment:
I don't quite like this test but mutant complained without it:

```
@event_store
        .read
        .stream("Taxes::AvailableVatRate$#{vat_rate_code}")
        .to_a
        .any?
  ```
 Mutant indicated that this code passes all the tests without the `stream` filter.

Reply:
I think the test if fine, but the name of it could be improved.

Reply:
I couldn't come up with a shorter name so I did something different :)
This and the previous test were basically testing the same thing, just the setup was slightly extended for this one. I extracted adding available VAT rate to the setup function and this way I was able to DRY up these tests a bit and also remove this last test (now the second test satisfies mutant as well)

Reply:
I had to change the functionality of VatRateCatalog so I removed these tests.